cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -3104,7 +3101,7 @@ std::optional<bool> CacheAllocator<CacheTrait>::saveNvmCache() {
     return false;
   }
 
-  nvmCacheState_.markSafeShutDown();
```

Comment:
Please add an `if` block to check `nvmCacheState_` has value. Otherwise do an XLOG(ERR, "Could not mark NvmCacheState as safe shutdown.") and return false;
This shouldn't happen but it's good to have for safety.

Reply:
Done

Reply:
Done

Reply:
@haowu14 I made the changes that you requested but I noticed that CI workflow didn't pass on my fork. I'll work on this a little more and post an update here.

Reply:
@haowu14 It turns out that the build is failing without my changes too.

Reply:
The failure that shows up in the build hook is the CLA. Please follow the instruction here https://github.com/facebook/CacheLib/pull/95#issuecomment-990067251 and after that I can import this for internal review. Thanks!